### DISCUSSION TITLE
Does have Next.js support for this library?

### DISCUSSION THREAD

**Discussion by devsin** [KevinVandy/material-react-table#78] [Q&A] [answered]
Because I tried this library on my current Next.js project and I got this following error: 
```
./node_modules/material-react-table/src/MaterialReactTable.tsx
Module parse failed: Unexpected token (8:12)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|   useMemo,
| } from 'react';
> import type {
|   AlertProps,
|   ButtonProps,
```

Also I tried on freshly created Next.js app and I got same error. 

And I tried with most basic configs: 
```
<MaterialReactTable columns={columns} data={data} />
```

Do you have any idea why I am getting this?

**Comment by KevinVandy** [comment, marked as answer]
Yes. My entire docs site is written in next.js, and I've never encountered this error. Will need to look into it.

**Reply by KevinVandy** [reply]
wait, how are you importing this?

**Reply by devsin** [reply]
😄 that's awkward. maybe I should delete the build files first. let me try it.

**Reply by devsin** [reply]
Like this. 
```
import MaterialReactTable from 'material-react-table/src/MaterialReactTable'
```

**Reply by devsin** [reply]
Yep this fixed the problem. 
```
import MaterialReactTable from 'material-react-table'
```

I have never give attention to imports. I am using WebStorm and I used IDE import suggestions. That's why I was getting this error. 

Thanks for your helps Kevin.

**Reply by KevinVandy** [reply]
Ok, you're just importing it wrong

`import MaterialReactTable from 'material-react-table'`

Don't import from src
